tfconfig/provider_ref.go

```diff
@@ -7,3 +7,17 @@ type ProviderRef struct {
 	Name  string `json:"name"`
 	Alias string `json:"alias,omitempty"` // Empty if the default provider configuration is referenced
 }
+
+type providersSortedByName []*ProviderRef
```

Comment:
I think this would be better as `type providerRefs`; they aren't sorted _yet_ and using the plural form for a type that encapsulates a list is a (loosely-held) convention around here. What do you think?

Reply:
you could hang the `Contains` method off of this type as well and keep the name